lib/public/views/Runs/format/formatRunDetectors.js

```diff
@@ -40,7 +40,7 @@ export const formatRunDetectors = (run, detailsModel) => {
             '.g2.flex-row.flex-grow.flex-wrap.justify-end',
             detectors.map(({ id, name }) => {
                 const quality = detectorsQualities.find((x) => x.id == id)
-                    ? detectorsQualities.find((x) => x.id == id).quality
+                    ? detectorsQualities.find((x) => x.id == id).quality || 'none'
```

Comment:
Here I would use an if, to make the code clearer because we start to combine a bit too much conditionals.

Here is my suggestion (replacing from l.42 to l.67)

```js
                let color;
                let icon;

                const detector = detectorsQualities.find((x) => x.id === id);

                // Run has the detector
                if (detector) {
                    const quality = detector.quality || null;

                    const actualQuality = detailsModel.isEditModeEnabled
                        ? detailsModel.runPatch.getDetectorQuality(id) || quality
                        : quality;

                    switch (actualQuality) {
                        case RunDetectorQualities.GOOD:
                            color = 'success';
                            icon = iconCheck();
                            break;
                        case RunDetectorQualities.BAD:
                            color = 'danger';
                            icon = iconX();
                            break;
                        // No quality specified for this detector
                        default:
                            color = 'gray-darker';
                            icon = iconBan();
                            break;
                    }
                // Run do not have the detector
                } else {
                    color = 'gray';
                    icon = iconBan();
                }
                
```